homeassistant/components/sensor/daikin.py

```diff
@@ -0,0 +1,98 @@
+"""
+Support for Daikin AC Sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.daikin/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_ICON,
+    CONF_MONITORED_CONDITIONS, CONF_TEMPERATURE_UNIT
+)
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.daikin import (
+    SENSOR_TYPES,
+    manual_device_setup
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=None): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=SENSOR_TYPES.keys()):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Daikin sensors."""
+    if discovery_info is not None:
+        host = discovery_info.get('ip')
+        name = None
+        monitored_conditions = discovery_info.get(
+            CONF_MONITORED_CONDITIONS, list(SENSOR_TYPES.keys())
+        )
+        _LOGGER.info("Discovered a Daikin AC sensor on %s", host)
+    else:
+        host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+        monitored_conditions = config.get(CONF_MONITORED_CONDITIONS)
+        _LOGGER.info("Added Daikin AC sensor on %s", host)
+
+    device = manual_device_setup(hass, host, name)
+
+    sensors = []
+    for monitored_state in monitored_conditions:
+        sensors.append(DaikinClimateSensor(device, name, monitored_state))
+
+    add_devices(sensors, True)
+
+
+class DaikinClimateSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, device, name=None, monitored_state=SENSOR_TYPES.keys()):
+        """Initialize the sensor."""
+        self._device = device
+        if name is None:
+            name = SENSOR_TYPES.get(monitored_state).get(CONF_NAME) \
```

Comment:
Does this `.get(…).get(…)` have a use? The difference between `.get(…)` and `[…]` is that `get(…)` does not give you a `KeyError` but returns `None` if the key is not found. However, in that case the second `.get(…)` will obviously fail?

Reply:
Let's go together on this:
- the monitored state is validate by the sensor PLATFORM_SCHEMA and allows only SENSOR_TYPES keys...so the 1st get should NEVER fail